PIM_BE/PIMTool/Services/ProjectService.cs

```diff
@@ -0,0 +1,411 @@
+    {
+        PagingDto result = new PagingDto();
+        int skip = (pageIndex - 1) * pageSize;
+        string sortOption = string.Empty;
+        if(sortNumber == "0" && sortName == "0" && sortStatus == "0" && sortCustomer == "0" && sortStartDate == "0")
+        {
+            sortNumber = ASC;
+        }
+        // Sort query
+        if (sortNumber != "0")
+        {
+            if (sortNumber == ASC)
+            {
+                sortOption = "ProjectNumber ASC";
+            }
+            else
+            {
+                sortOption = "ProjectNumber DESC";
+            }
+
+        }
+        else if (sortName != "0")
+        {
+            if (sortName == ASC)
+            {
+                sortOption = "Name ASC";
+            }
+            else
+            {
+                sortOption = "Name DESC";
+            }
+        }
+        else if (sortStatus != "0")
+        {
+            if (sortStatus == ASC)
+            {
+                sortOption = "Status ASC";
+            }
+            else
+            {
+                sortOption = "Status DESC";
+            }
+        }
+        else if (sortCustomer != "0")
+        {
+            if (sortCustomer == ASC)
+            {
+                sortOption = "Customer ASC";
+            }
+            else
+            {
+                sortOption = "Customer DESC";
+            }
+        }
+        else if (sortStartDate != "0")
+        {
+            if (sortStartDate == ASC)
+            {
+                sortOption = "StartDate ASC";
+            }
+            else
+            {
+                sortOption = "StartDate DESC";
+            }
+        }
+        else
+        {
+            sortOption = null;
+        }
+
+        // Search query
+        string queryString = string.Empty;
+        if (string.IsNullOrEmpty(searchText) && searchStatus.Equals("0"))
+        {
+            queryString = null;
+        }
+        else if (string.IsNullOrEmpty(searchText))
+        {
+            queryString = $"Status.Equals(\"{searchStatus}\")";
+        }
+        else if (searchStatus.Equals("0"))
+        {
+            //queryString = $"Name LIKE '%{searchText}%' or Customer LIKE '%{searchText}%'";
+            queryString = $"Name.Contains(\"{searchText}\") || Customer.Contains(\"{searchText}\") || (ProjectNumber.ToString().Contains(\"{searchText}\"))";
+        }
+        else
+        {
+            queryString = $"((Name.Contains(\"{searchText}\") || Customer.Contains(\"{searchText}\"))  || (ProjectNumber.ToString().Contains(\"{searchText}\"))) && Status.Equals(\"{searchStatus}\")";
+        }
+        // Execute query
+        if (queryString != null && sortOption != null)
+        {
+            var temp = _repository.Get().Where(queryString).OrderBy(sortOption);
```

Comment:
[C] The code can not handle for general case.
We also use RawSQL without param which can lead SQL injection